### PR TITLE
V2 role based access

### PR DESCRIPTION
**Pull request by sijumoncy**
#250 
This PR contains only auth and access check for autographa project management and not for translation APIs.

### CODE REVIEW

File: app/authentication.py [hunk 1]
```diff
@@ -279,26 +280,65 @@ def filter_resource_content_get(db_resource, access_tags, required_permission, u
...
+        if len(filtered_content) == 0:
+            for project in db_resource:
+                project_user_obj = project.users
+                for user in project_user_obj:
+                    if user_id == user.userId and\
+                        user.userRole in allowed_users and \
+                            not any(project == dic for dic in filtered_content):
+                        filtered_content.append(project)
+
+    if len(filtered_content) > 0:
```
Comment (kavitharaju) [inline app/authentication.py:313]: What happen if this API is called to fill the home page of a  new user who doesn't have any projects? Will this throw an error?

File: app/authentication.py [hunk 2]
```diff
@@ -478,15 +527,22 @@ def kratos_logout(self,auth:HTTPAuthorizationCredentials= Security(security)):
         return data
 
 #get all user details
-def get_all_kratos_users():
-    """get all user info"""
+def get_all_kratos_users(rec_user_id=None):
```
Comment (kavitharaju) [inline app/authentication.py:530]: better to remove "all" from the function name as it is used to get either all or a specific user now.

Comment (sijumoncy) [reply app/authentication.py]: ok

File: app/authentication.py (hunk 1 above)
Comment (sijumoncy) [reply app/authentication.py]: yes currently it will throw an error only .But we can change it to [ ]

File: app/test/test_agmt_projects.py [hunk 3]
```diff
@@ -775,20 +775,17 @@ def test_get_project_access_rules():
...
     headers_auth['Authorization'] = "Bearer"+" "+initial_test_users['APIUser']['token']
     response = client.get(UNIT_URL,headers=headers_auth)
-    assert response.status_code == 403
-    assert response.json()['error'] == 'Permision Denied'
+    assert len(response.json()) == 0
     headers_auth['Authorization'] = "Bearer"+" "+initial_test_users['VachanUser']['token']
     response = client.get(UNIT_URL,headers=headers_auth)
-    assert response.status_code == 403
-    assert response.json()['error'] == 'Permision Denied'
+    assert len(response.json()) == 0
```
Comment (kavitharaju) [inline app/test/test_agmt_projects.py:788]: There is a check for notAvailableContent you could use here. Just letting you know for future cases.

**Review by kavitharaju — APPROVED** [review]

**Pull request merged**